syncer/syncer.go

```diff
@@ -1685,6 +1741,11 @@ func (s *Syncer) Run(ctx context.Context) (err error) {
 		startTime := time.Now()
 		e, err = s.getEvent(tctx, currentLocation)
 
+		if waitXIDStatus(s.waitXIDJob.Load()) == waitComplete {
+			s.tctx.L().Info("All event is completed before syncer close, the coming event will be reject", zap.Any("event", e))
+			return nil
```

Comment:
this return may trigger the `defer runCancel()` in line 1448, so `case <-runCtx.Done():` in line 1470 is entered and we have no time to wait job executed. (we change `s.waitXIDJob` to `waitComplete` based on receiving the XID event, at that time previous DML jobs is not executed) 🤔

Reply:
We will wait job in L1639 first, then `defer runCancel()`